cluster-operator/src/test/java/io/strimzi/operator/cluster/model/CruiseControlTest.java

```diff
@@ -0,0 +1,372 @@
+
+    public void checkOwnerReference(OwnerReference ownerRef, HasMetadata resource)  {
+        assertThat(resource.getMetadata().getOwnerReferences().size(), is(1));
+        assertThat(resource.getMetadata().getOwnerReferences().get(0), is(ownerRef));
+    }
+
+    private Map<String, String> expectedLabels(String name)    {
+        return TestUtils.map(Labels.STRIMZI_CLUSTER_LABEL, this.cluster,
+                "my-user-label", "cromulent",
+                Labels.STRIMZI_KIND_LABEL, Kafka.RESOURCE_KIND,
+                Labels.STRIMZI_NAME_LABEL, name,
+                Labels.KUBERNETES_NAME_LABEL, Labels.KUBERNETES_NAME,
+                Labels.KUBERNETES_INSTANCE_LABEL, this.cluster,
+                Labels.KUBERNETES_MANAGED_BY_LABEL, AbstractModel.STRIMZI_CLUSTER_OPERATOR_NAME);
+    }
+
+    private Map<String, String> expectedSelectorLabels()    {
+        return Labels.fromMap(expectedLabels()).strimziLabels().toMap();
+    }
+
+    private Map<String, String> expectedLabels()    {
+        return expectedLabels(CruiseControlResources.deploymentName(cluster));
+    }
+
+    private List<EnvVar> getExpectedEnvVars() {
+        List<EnvVar> expected = new ArrayList<>();
+        expected.add(new EnvVarBuilder().withName(CruiseControl.ENV_VAR_STRIMZI_KAFKA_BOOTSTRAP_SERVERS).withValue(CruiseControl.defaultBootstrapServers(cluster)).build());
+        expected.add(new EnvVarBuilder().withName(KafkaMirrorMakerCluster.ENV_VAR_STRIMZI_KAFKA_GC_LOG_ENABLED).withValue(Boolean.toString(AbstractModel.DEFAULT_JVM_GC_LOGGING_ENABLED)).build());
+        expected.add(new EnvVarBuilder().withName(CruiseControl.ENV_VAR_MIN_INSYNC_REPLICAS).withValue(minInsyncReplicas).build());
+        expected.add(new EnvVarBuilder().withName(KafkaMirrorMakerCluster.ENV_VAR_KAFKA_HEAP_OPTS).withValue(kafkaHeapOpts).build());
+        expected.add(new EnvVarBuilder().withName(CruiseControl.ENV_VAR_CRUISE_CONTROL_CONFIGURATION).withValue(ccConfig.getConfiguration()).build());
+
+        return expected;
+    }
+
+    @Test
+    public void testFromConfigMap() {
+        assertThat(cc.namespace, is(namespace));
+        assertThat(cc.cluster, is(cluster));
+        assertThat(cc.getImage(), is(ccImage));
+    }
+
+    @Test
+    public void testGenerateDeployment() {
+        Deployment dep = cc.generateDeployment(null, true, null, null);
+
+        List<Container> containers = dep.getSpec().getTemplate().getSpec().getContainers();
+
+        assertThat(containers.size(), is(2));
+
+        assertThat(dep.getMetadata().getName(), is(CruiseControlResources.deploymentName(cluster)));
+        assertThat(dep.getMetadata().getNamespace(), is(namespace));
+        assertThat(dep.getMetadata().getOwnerReferences().size(), is(1));
+        assertThat(dep.getMetadata().getOwnerReferences().get(0), is(cc.createOwnerReference()));
+
+        // checks on the main Cruise Control container
+        assertThat(containers.get(0).getImage(), is(cc.image));
```

Comment:
You should avoid using the index to get the container as it could (theoretically) be a different one to the one you expect. You can grab them by name by supplying a predicate that checks for the name.